tests/foreman/api/test_subscription.py

```diff
@@ -385,30 +385,118 @@ def test_positive_expired_SCA_cert_handling(module_sca_manifest_org, rhel_conten
-        2. Set Restriction to OS on repos
+        1. Check all repos enabled from start (unrestricted)
+        2. Set Restriction to OS (RHEL10/9/8) on each repo (hammer repository update)
         3. Subscription-manager refresh
-        4. Verify enabled repos
+        4. Verify available repo(s) for content host using subscription-manager.
 
-    :expectedresults: Custom EPEL repos with OS restrictions set are
-        disabled based on its corresponding RHEL version
+    :expectedresults:
+        - One custom repo available for client, with matching OS restriction.
+        - Two custom repos not available for client, with non-matching OS restrictions.
 
     :customerscenario: true
 
     :BZ: 1526564
 
     :CaseImportance: High
 
-    :CaseAutomation: NotAutomated
     """
+    org = module_sca_manifest_org
+    # 3 newest supported rhel versions from tail, exclude fips
+    rhel_versions = [
+        rhel_major_ver
+        for rhel_major_ver in settings.supportability.content_hosts.rhel.versions
+        if 'fips' not in str(rhel_major_ver)
+    ][-3:]
+    # create & sync 3 custom repositories
+    repo_urls = [settings.repos.yum_9.url, settings.repos.yum_6.url, settings.repos.yum_1.url]
+    custom_repos = []
+    for url in repo_urls:
+        r = target_sat.api.Repository(
+            url=url,
+            product=module_product,
+        ).create()
+        r.sync()
+        custom_repos.append(r.read())
+    # add the repos to module CV
+    module_cv.repository = custom_repos
+    module_cv.update(['repository'])
+    module_cv = module_cv.read()
+    # publish CV, associate AK, register host and enable repos
+    setup = target_sat.api_factory.register_host_and_needed_setup(
+        organization=org,
+        client=rhel_contenthost,
+        activation_key=module_ak,
+        environment='Library',
+        content_view=module_cv,
+        enable_repos=True,
+    )
+    assert setup['result'] != 'error', f'{setup["message"]}'
+    # registered host returned by setup
+    assert (client := setup['client'])
+    result = client.execute('subscription-manager refresh')
+    assert result.status == 0, f'{result.stdout}'
+
+    # get available repos from sub-man repos list
+    sub_man_repos = client.subscription_manager_list_repos().stdout
+    # assert all 3 repos are available at start (unrestricted)
+    for repo in custom_repos:
+        assert repo.label in sub_man_repos
+
+    # restrict each repo to a different RHEL major version
+    rhel_majors = [ver for ver in rhel_versions]
+    matching_repo_id = None
+    for r in custom_repos:
+        repo_rhel = rhel_majors[-1]
+        # restrict repo to OS via hammer, ie 'rhel-9'
+        formatted = 'rhel-' + str(repo_rhel)
+        result = target_sat.execute(
+            f'hammer repository update --os-versions "{formatted}"'
+            f' --id {r.id} --organization-id {org.id}'
+        )
+        assert result.status == 0, f'{result.stdout}'
```

Comment:
```suggestion
        target_sat.cli.Repository.update({'os-version': formatted, 'id': r.id})
```
should throw exception for non-zero result

Reply:
whoops i believe it should be `os-versions` plural. Let me give that a try!